lib/rmt/lockfile.rb

```diff
@@ -4,13 +4,13 @@ class RMT::Lockfile
 
   class << self
     def create_file
-      if File.exist?(LOCKFILE_LOCATION)
-        raise RMT::Lockfile::ExecutionLockedError
-      else
-        dirname = File.dirname(LOCKFILE_LOCATION)
-        FileUtils.mkdir_p(dirname) unless File.directory?(dirname)
-        File.open(LOCKFILE_LOCATION, 'w') { |f| f.write(Process.pid) }
-      end
+      dirname = File.dirname(LOCKFILE_LOCATION)
+      FileUtils.mkdir_p(dirname) unless File.directory?(dirname)
+
+      # has to be a class instance variable otherwise garbage-collected and lock is removed
+      @f = File.open(RMT::Lockfile::LOCKFILE_LOCATION, File::RDWR | File::CREAT)
```

Comment:
Can you please provide a comment or link that explains this lines. It's far from obvious to me.

Reply:
Maybe a link to https://ruby-doc.org/core-2.3.1/File.html#method-i-flock would suffice. The code is pulled straight from the example there.